lib/id3c/api/routes.py

```diff
@@ -145,6 +146,99 @@ def receive_fhir(*, session):
     return "", 204
 
 
+@api_v1.route("/warehouse/identifier/<id>", methods = ['GET'])
+@authenticated_datastore_session_required
+def get_identifier(id, *, session):
+    """
+    Retrieve an identifier's metadata.
+
+    GET /warehouse/identifier/*id* to receive a JSON object containing the
+    identifier's record.  *id* may be a full UUID or shortened barcode.
+    """
+    LOG.debug(f"Fetching identifier «{id}»")
+
+    identifier = datastore.fetch_identifier(session, id)
+
+    return jsonify(identifier._asdict())
+
+
+@api_v1.route("/warehouse/identifier-sets", methods = ['GET'])
+@authenticated_datastore_session_required
+def get_identifier_sets(*, session):
+    """
+    Retrieve metadata about all identifier sets.
+
+    GET /warehouse/identifier-set to receive a JSON array of objects, each
+    containing a set's metadata fields.
+    """
+    LOG.debug(f"Fetching identifier sets")
+
+    sets = datastore.fetch_identifier_sets(session)
+
+    return jsonify([ set._asdict() for set in sets ])
+
+
+@api_v1.route("/warehouse/identifier-sets/<name>", methods = ['GET'])
+@authenticated_datastore_session_required
+def get_identifier_set(name, *, session):
+    """
+    Retrieve an identifier set's metadata.
+
+    GET /warehouse/identifier-set/*name* to receive a JSON object containing the set's
+    metadata fields.
+    """
+    LOG.debug(f"Fetching identifier set «{name}»")
+
+    set = datastore.fetch_identifier_set(session, name)
+
+    return jsonify(set._asdict())
+
+
+@api_v1.route("/warehouse/identifier-sets/<name>", methods = ['PUT'])
+@authenticated_datastore_session_required
+def put_identifier_set(name, *, session):
+    """
+    Make a new identifier set.
+
+    PUT /warehouse/identifier-sets/*name* to create the set if it doesn't yet
+    exist.  201 Created is returned when the set is created, 204 No Content if
+    the set already existed.
+    """
+    LOG.debug(f"Making identifier set «{name}»")
+
+    new_set = datastore.make_identifier_set(session, name)
+
+    return "", 201 if new_set else 204
+
+
+@api_v1.route("/warehouse/identifier-sets/<name>/identifiers", methods = ['POST'])
```

Comment:
Is it worth considering setting an upper limit to n value here?

Reply:
Ah, good call. It's always good to consider what limits are appropriate. I'm not sure what reasonable bound would be though… Here's the current summary stats of batch size for all we've minted so far:

```
min	     1
mode	 1,040
max	11,900
```

Stepping back from `n`, I think what a limit would really seek to prevent is minting reqs that run "forever". We can't guarantee this just through `n` alone (although very small `n` might approximate it well enough), so maybe the more appropriate thing would be an explicit time limit. There is some existing limit on request/response duration already at the web server level (both at uWSGI and Apache, I believe), so that may provide enough of a time limit if the corresponding database session is terminated when the request is. But I'm not sure that's the case.

This also means that we may not be able to use this API as-is to mint large batches of identifiers without increasing existing request/response duration limits, possibly to unreasonable levels. An async API would obviate that issue, but then require a minting job queue and polling of job status by the client. (We've talked about this in the past in related convos.)

@seattleflu/core-devs thoughts?

Reply:
I think it makes sense to drop it or if possible just disable that route until there's a use case for it. I see that the `id3c identifier mint` command is used when generating barcode labels, but I'm not familiar enough with all of the workflows to know which ones might need this functionality.

Reply:
Yeah, I agree, reasonable to drop it from this PR. I'd move it to another branch and push that up as a wip, to potentially be picked up by someone else in the future. Would love input from others here too before I do.

Reply:
Repushed dropping the minting endpoint.